File: LeetCode/217-Contains.Duplicate/solution.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        visited = set() # used for keeping track of UNIQUE values

        # go through every number of input array
        for num in nums:
            # return True if we've seen the current element before
            if num in visited:
                return True
            # otherwise add it to elements we've seen before
            visited.add(num)

        # womp womp -- no duplicates!
        return False

File: LeetCode/217-Contains.Duplicate/solution.java
public boolean containsDuplicate(int[] nums) {
    // for tracking the unique elements
    Set<Integer> set = new HashSet<>(nums.length);

    // iterate through each element
    for (int x: nums) {
        // if the element is already in the set, return true
        if (set.contains(x)) return true;
        // otherwise, add the element to the set
        set.add(x);
    }
    // womp womp
    return false;
}